### ISSUE TITLE
Getting item in Draft status when using fields options in api call	

### ISSUE THREAD

**Issue by trotin** [pauloamgomes/CockpitCMS-Moderation#4]
I'm fan of your addon. It is very helpfull when creating blog site.
But It seems to have an issue in certain circonstance.
Retrieving a collection by simple GET API call (without any option),  Unpublished and Draft items are filtered and only published items are displayed. Fine !
But calling the API using POST method to specify some fields to be displayed, Published and Draft items are returned.
Using other option (like sort, limit or filter) on POST call, do not cause similar problem

I'm be carefull to talk only about just created items. They have never been in any other status before.

**Comment by trotin** [comment]
Diving deeper in your source code, I found an explanation. We need to include the moderation field in the list of fields of the POST request to activate the feature. So by not adding that field, we can 'bypass' Draft filter.
For unpublished items, the filter works in any way to request the API (with or without specify fields)

**Comment by pauloamgomes** [comment]
Hi @trotin , yes, thats true, the unpublished state is handled on `collections.find.before` event so it will work always. The draft needs to be handled on the `collections.find.after` event as need to iterate over the revisions to get the last published version (before the draft).

Need to do some tests to confirm that adding the moderation field in the options will work well, in the meantime other option is to implement the `moderation.find.before` event and add there the field, e.g.:

```
$app->on('moderation.find.before', function($name, &$options) {
  $options['fields']['status'] = 1;
});
```

so in theory you don't need to include the field in your request as it will included by above event.

**Comment by pauloamgomes** [comment]
Fixed by [cd1e641](https://github.com/pauloamgomes/CockpitCMS-Moderation/commit/cd1e64133a97b1bce6d8a2cd4e79f6e9c4f39670)
